include/SDL3/SDL_gpu.h

```diff
@@ -3674,6 +3675,25 @@ extern SDL_DECLSPEC bool SDLCALL SDL_CancelGPUCommandBuffer(
 extern SDL_DECLSPEC bool SDLCALL SDL_WaitForGPUIdle(
     SDL_GPUDevice *device);
 
+/**
+ * Blocks the thread until a swapchain texture is available to be acquired.
+ * Calling this function before SDL_AcquireGPUSwapchainTexture() will ensure
+ * that the acquire will immediately return a swapchain texture under most, but
+ * not all, conditions.
+ *
+ * \param device a GPU context.
+ * \param window a window that has been claimed.
+ * \returns true on success, false on failure; call SDL_GetError() for more
+ *          information.
+ *
+ * \since This function is available since SDL 3.2.0.
+ *
+ * \sa SDL_AcquireGPUSwapchainTexture
+ */
+extern SDL_DECLSPEC bool SDLCALL SDL_WaitForGPUSwapchain(
```

Comment:
I don't think so, because I can imagine a low-latency applications might want to wait for the swapchain to be available but delay actually acquiring it until after input processing, for example. But maybe @lmurray can comment more on that.

Reply:
For throughput-optimized apps any waiting for the swapchain should occur immediately before acquiring the swapchain texture as this gives the GPU the most amount of time to complete its work without stalling the CPU. Since SDL_AcquireGPUSwapchainTexture() already waits or queries the exact same fence it is mildly wasteful to use the proposed SDL_WaitForGPUSwapchain() immediately before this function and merging the functionality together is useful.

For latency-optimized apps any waiting for the swapchain should occur sometime after submitting the command buffer that acquired the previous swapchain texture but before processing any events received by input devices. This means that any input events that were received during the wait are able to be included in the next frame to be rendered. Since processing input events typically also requires processing other system events such as window resizing acquiring the swapchain texture at this time is risky as if the swapchain is invalidated then any usage of the returned swapchain texture will be wasteful as the result will be discarded when submitted. For this purpose you want the wait to be a separate call.

Ideally I think SDL_AcquireGPUSwapchainTexture() should always just query the fence with no waiting even when using the FIFO presentation mode but changing that now will break any app that depends on the internal wait existing.

Reply:
If that's the case I actually think we should just change SDL_AcquireGPUSwapchainTexture like you stated. We've already made it clear everywhere that you have to check if the swapchain texture is null when acquiring so that change shouldn't break anything. I think making the blocking behavior explicit rather than implicit is probably for the best.

Reply:
To summarize I think we should make the following changes:

- Add SDL_WaitForGPUSwapchain() which blocks until a swapchain texture is available
- Change SDL_AcquireGPUSwapchainTexture() to immediately return false if a swapchain texture is unavailable regardless of present mode
- Add SDL_WaitAndAcquireGPUSwapchainTexture() which blocks until a swapchain texture is available and then acquires it